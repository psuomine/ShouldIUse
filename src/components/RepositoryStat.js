import React from 'react'
import styled from 'styled-components'
import { H3, H2 } from 'components/styles/Heading'

const Container = styled.div`
  display: flex;
  width: 300px;
  flex-direction: column;
  align-items: center;
`

const RepositoryStat = ({ title, value, unit }) => {
  return (
    <Container>
      <H3 data-testid="stat-title">{title}</H3>
      <H2 data-testid="stat-value">{value}</H2>
    </Container>
  )
}

export default RepositoryStat
