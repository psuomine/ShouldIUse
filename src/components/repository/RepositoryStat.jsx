import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { H1 } from 'components/styles/Heading'
import { BodyText } from 'components/styles/Typography'
import StatusIcon from 'components/StatusIcon'

const Container = styled.div`
  position: relative;
  display: flex;
  width: 280px;
  height: 245px;
  flex-direction: column;
  align-items: center;
  margin-bottom: 24px;
  background-color: white;
  border-radius: 4px;
  padding: 24px;
  justify-content: center;
`

const Value = styled(H1)`
  margin-top: 32px;
  font-weight: 700;
  letter-spacing: 2.5px;
`

const StatusIconAbsolute = styled(StatusIcon)`
  position: absolute;
  top: 24px;
`

const RepositoryStat = ({ title, value, isValid }) => (
  <Container>
    <StatusIconAbsolute isValid={isValid} />
    <Value data-testid="stat-value" color="primary">
      {value}
    </Value>
    <BodyText data-testid="stat-title" color="muted">
      {title}
    </BodyText>
  </Container>
)

RepositoryStat.propTypes = {
  title: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  isValid: PropTypes.bool.isRequired,
}

export default RepositoryStat
