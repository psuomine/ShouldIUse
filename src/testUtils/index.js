import React from 'react'
import { render } from 'react-testing-library'
import { ThemeProvider } from 'styled-components'
import Theme from 'theme/Theme'

const customRender = (node, options) => render(<ThemeProvider theme={Theme}>{node}</ThemeProvider>, options)

// re-export everything
export * from 'react-testing-library'

// override render method
export { customRender as render }
