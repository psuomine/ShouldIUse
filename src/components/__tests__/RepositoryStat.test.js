import React from 'react'
import { render } from 'testUtils'
import RepositoryStat from '../Repository/RepositoryStat'

describe('<SearchCard />', () => {
  it('renders prop values', () => {
    const { getByTestId } = render(
      <RepositoryStat title="lodash" value="8" isValid />
    )
    expect(getByTestId('stat-title').textContent).toBe('lodash')
    expect(getByTestId('stat-value').textContent).toBe('8')
  })

  it('renders success icon if isValid is true, snapshot', () => {
    const { container } = render(
      <RepositoryStat title="lodash" value="8" isValid />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders error icon if isValid is true', () => {
    const { container } = render(
      <RepositoryStat title="lodash" value="8" isValid={false} />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
