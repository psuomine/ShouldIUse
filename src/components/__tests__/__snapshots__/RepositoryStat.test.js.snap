// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchCard /> renders error icon if isValid is true 1`] = `
<div
  class="sc-gzVnrw fiCOXn"
>
  <div
    class="sc-bZQynM cZJhhU"
  >
    <svg
      class="sc-EHOje ibVUnr"
      color="rgb(250,71,79)"
      data-testid="svg-icon"
      height="24"
      width="24"
    >
      <path
        d="M24 2.41714L21.5829 0L12 9.58286L2.41714 0L0 2.41714L9.58286 12L0 21.5829L2.41714 24L12 14.4171L21.5829 24L24 21.5829L14.4171 12L24 2.41714Z"
        data-testid="svg-icon-path"
      />
    </svg>
  </div>
  <h1
    class="sc-bdVaJa sc-htoDjs hbgios"
    color="primary"
    data-testid="stat-value"
  >
    8
  </h1>
  <p
    class="sc-bxivhb dzllQf"
    color="muted"
    data-testid="stat-title"
    font-weight="400"
  >
    lodash
  </p>
</div>
`;

exports[`<SearchCard /> renders success icon if isValid is true, snapshot 1`] = `
<div
  class="sc-gzVnrw fiCOXn"
>
  <div
    class="sc-bZQynM gpLXoD"
  >
    <svg
      class="sc-EHOje hfLPRj"
      color="rgb(23,195,123)"
      data-testid="svg-icon"
      height="18"
      width="24"
    >
      <path
        d="M7.42449 14.5737L2.42002 9.56926C1.86238 9.01162 0.975871 9.01162 0.418231 9.56926C-0.13941 10.1269 -0.13941 11.0134 0.418231 11.571L6.40929 17.5621C6.96693 18.1197 7.86774 18.1197 8.42538 17.5621L23.5818 2.42002C24.1394 1.86238 24.1394 0.975871 23.5818 0.418231C23.0241 -0.13941 22.1376 -0.13941 21.58 0.418231L7.42449 14.5737Z"
        data-testid="svg-icon-path"
      />
    </svg>
  </div>
  <h1
    class="sc-bdVaJa sc-htoDjs hbgios"
    color="primary"
    data-testid="stat-value"
  >
    8
  </h1>
  <p
    class="sc-bxivhb dzllQf"
    color="muted"
    data-testid="stat-title"
    font-weight="400"
  >
    lodash
  </p>
</div>
`;
